openpype/hosts/nuke/api/lib.py

```diff
@@ -2114,12 +2141,87 @@ def set_root_colorspace(self, imageio_host):
 
 Reopening Nuke should synchronize these paths and resolve any discrepancies.
 """
-                nuke.message(
-                    message.format(
-                        env_path=current_ocio_path,
-                        settings_path=config_data["path"]
-                    )
+            nuke.message(
+                message.format(
+                    env_path=current_ocio_path,
+                    settings_path=config_data["path"]
                 )
+            )
+            return True
+
+    def _set_ocio_config_path_to_workfile(self, config_data):
+        """ Set OCIO config path to workfile
+
+        Path set into nuke workfile. It is trying to replace path with
+        environment variable if possible. If not, it will set it as it is.
+        It also saves the script to apply the change, but only if it's not
+        empty Untitled script.
+
+        Args:
+            config_data (dict): OCIO config data from settings
+
+        """
+        # replace path with env var if possible
+        ocio_path = self._replace_ocio_path_with_env_var(
```

Comment:
I guess that most of studio will use their custom environment variables, and because `_replace_ocio_path_with_env_var` count with only default env variables, I would suggest to use "raw" ocio path from settings, instead of trying to find possible environment variable in already filled path, or maybe just use `OCIO` env variable (which is filled in global prelaunc hook).

That would break only if settings for OCIO path would change during production, I don't know if we should allow that?

NOTE: I've also seen settings for OCIO withou any env varible (full path for each platform).

Reply:
yes, thanks for suggestion. It is processed.